File: src/main/java/com/slavikjunior/dao/UserDAO.java
package com.slavikjunior.dao;

import java.sql.*;

public class UserDAO {

    private Connection conn = getConnection();

    private Connection getConnection() {
        try {
            Class.forName("org.postgresql.Driver");
        } catch (ClassNotFoundException e) {
            System.out.println("Driver not found");
        }
        try {
            Connection conn = DriverManager.getConnection(
                    "jdbc:postgresql://localhost:5432/demo",
                    "postgres",
                    "7913"
            );
            System.out.println("Connected to PostgreSQL database successfully");
            return conn;
        } catch (SQLException e) {
            e.printStackTrace();
            return null;
        }
    }


    public boolean userExists(String username) throws SQLException {
        PreparedStatement ps = conn.prepareStatement("SELECT 1 FROM users WHERE username=?");
        ps.setString(1, username);
        ResultSet rs = ps.executeQuery();
        return rs.next();
    }

    public void createUser(String username, String password, String email) throws SQLException {
        PreparedStatement ps = conn.prepareStatement("INSERT INTO users (username, password, email) VALUES (?, ?, ?)");
        ps.setString(1, username);
        ps.setString(2, password);
        ps.setString(3, email);
        ps.executeUpdate();
    }

    public boolean checkCredentials(String username, String password) throws SQLException {
        PreparedStatement ps = conn.prepareStatement("SELECT * FROM users WHERE username=? AND password=?");
        ps.setString(1, username);
        ps.setString(2, password);
        ResultSet rs = ps.executeQuery();
        return rs.next();
    }
}

File: src/main/java/com/slavikjunior/dao/UserDao.kt
package com.slavikjunior.dao

import com.slavikjunior.db_manager.getConnection
import java.sql.Connection
import java.sql.PreparedStatement
import java.sql.SQLException

private const val USER_DATABASE_NAME = "users"
val connection = getConnection(USER_DATABASE_NAME)

// todo почитать про исключения в котлине
//@Throws(SQLException::class)
fun userIsExists(login: String): Boolean {
    if (!isConnectionEstablished())
        throw SQLException()
    connection as Connection
    val ps: PreparedStatement = connection.prepareStatement("SELECT 1 FROM users WHERE login=?")
    ps.setString(1, login)
    val rs = ps.executeQuery()
    return rs.next()
}

//@Throws(SQLException::class)
fun createUser(login: String, password: String, email: String) {
    if (!isConnectionEstablished())
        throw SQLException()
    connection as Connection
    val ps: PreparedStatement = connection.prepareStatement("INSERT INTO users (login, password, email) VALUES (?, ?, ?)")
    ps.setString(1, login)
    ps.setString(2, password)
    ps.setString(3, email)
    ps.executeUpdate()
}

//@Throws(SQLException::class)
fun checkCredentials(username: String, password: String): Boolean {
    if (!isConnectionEstablished())
        throw SQLException()
    connection as Connection
    val ps: PreparedStatement = connection.prepareStatement("SELECT * FROM users WHERE username=? AND password=?")
    ps.setString(1, username)
    ps.setString(2, password)
    val rs = ps.executeQuery()
    return rs.next()
}

private fun isConnectionEstablished() = if (connection != null) true else false